src/pool/srv_target.c

```diff
@@ -380,7 +380,7 @@ pool_free_ref(struct daos_llink *llink)
 
 	D_DEBUG(DF_DSMS, DF_UUID": freeing\n", DP_UUID(pool->sp_uuid));
 
-	ds_iv_ns_destroy(pool->sp_iv_ns);
+	ds_iv_ns_put(pool->sp_iv_ns);
```

Comment:
So it will be possible that the crt group and IV NS still have references but the ds_pool object has been freed?

Hmm, maybe we should move IV operations' retry logic from iv_op to callers' code, so that they are easier to abort (e.g., rebuild_leader_status_check will be able to check rgt->rgt_abort before retrying rebuild_iv_update)? Then we will not need these IV NS reference counting and start/stop/get/put, because ds_pool already has these functionality.

Reply:
IV NS will be freed together with ds_pool object, so it is impossible for IV NS to become an orphan.  see ds_iv_ns_stop().

Well, there are other use cases for IV operation. I think we need reference count for IVNS,  otherwise I do not see how the "orphan" IVNS can be avoided during pool destory process,  unless there is global barrier, pre-destory etc.

Probably retry logic can be moved to each individual operation, but I do not see much benefits for this issue, and I will check later.